cadctl/cmd/cluster-missing/cluster-missing.go

```diff
@@ -29,45 +30,68 @@ import (
-			return fmt.Errorf("one of the required envvars in the list '(AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN AWS_DEFAULT_REGION)' is missing")
-		}
-
-		_, err := aws.NewClient(AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN, AWS_DEFAULT_REGION)
-		if err != nil {
-			return fmt.Errorf("could not start awsClient: %w", err)
-		}
-
-		// in this case it's ok if the envvar is empty
-		CAD_OCM_FILE_PATH := os.Getenv("CAD_OCM_FILE_PATH")
-		_, err = ocm.New(CAD_OCM_FILE_PATH)
-		if err != nil {
-			return fmt.Errorf("could not create ocm client: %w", err)
-		}
-
-		CAD_PD, ok := os.LookupEnv("CAD_PD")
-		if !ok {
-			return fmt.Errorf("could not load CAD_PD envvar")
-		}
-
-		_, err = pagerduty.NewWithToken(CAD_PD)
-		if err != nil {
-			return fmt.Errorf("could not start pagerdutyClient: %w", err)
-		}
-
-		return nil
-	},
+	RunE:  run,
+}
+
+func run(cmd *cobra.Command, args []string) error {
+
+	AWS_ACCESS_KEY_ID, hasAWS_ACCESS_KEY_ID := os.LookupEnv("AWS_ACCESS_KEY_ID")
+	AWS_SECRET_ACCESS_KEY, hasAWS_SECRET_ACCESS_KEY := os.LookupEnv("AWS_SECRET_ACCESS_KEY")
+	AWS_SESSION_TOKEN, hasAWS_SESSION_TOKEN := os.LookupEnv("AWS_SESSION_TOKEN")
+	AWS_DEFAULT_REGION, hasAWS_DEFAULT_REGION := os.LookupEnv("AWS_DEFAULT_REGION")
+	if !hasAWS_ACCESS_KEY_ID || !hasAWS_SECRET_ACCESS_KEY || !hasAWS_SESSION_TOKEN || !hasAWS_DEFAULT_REGION {
+		return fmt.Errorf("one of the required envvars in the list '(AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN AWS_DEFAULT_REGION)' is missing")
+	}
+
+	awsClient, err := aws.NewClient(AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN, AWS_DEFAULT_REGION)
+	if err != nil {
+		return fmt.Errorf("could not start awsClient: %w", err)
+	}
+
+	// in this case it's ok if the envvar is empty
+	CAD_OCM_FILE_PATH := os.Getenv("CAD_OCM_FILE_PATH")
+	ocmClient, err := ocm.New(CAD_OCM_FILE_PATH)
+	if err != nil {
+		return fmt.Errorf("could not create ocm client: %w", err)
+	}
+
+	CAD_PD, ok := os.LookupEnv("CAD_PD")
+	if !ok {
+		return fmt.Errorf("could not load CAD_PD envvar")
+	}
+
+	_, err = pagerduty.NewWithToken(CAD_PD)
+	if err != nil {
+		return fmt.Errorf("could not start pagerdutyClient: %w", err)
+	}
+
+	chgmClient := chgm.Client{
+		ChgmService: chgm.ChgmProvider{
+			AwsClient: awsClient,
+			OcmClient: ocmClient,
+		},
+	}
+
+	res, err := chgmClient.AreAllInstancesRunning(externalClusterID)
+	if err != nil {
+		return fmt.Errorf("could not verify all instances were running: %w", err)
+	}
+	fmt.Printf("yay! the result is %#v\n", res)
+
+	return nil
 }
 
 var (
-	incidentID string
+	// externalClusterID has a value although it's the default to denote that this is the default value, if changed it will be the default value for externalClusterID_flagName
+	externalClusterID string = ""
 )
 
 func init() {
-	ClusterMissingCmd.Flags().StringVarP(&incidentID, "payload", "p", "", "The incident payload as received from the PagerDuty WebHook")
```

Comment:
@georgettica  why did you remove the incident payload?

Reply:
just because I didn't use it now.
currently the entrypoint is externalid, once we write code that ties from payload to externalid I will add it here.

also, it's easier to test when you provide the externalid

Reply:
makes sense, thanks for the clarification.

Reply:
keeping this open as this might be a question others reviewers might ask

Reply:
wasn't asked :/
not sure if I want to close it though